### PR TITLE
A little help

### PR DESCRIPTION
**Pull request by gleesik**
Fixed the recursive call in gulp tasks. Added npm start script. Created a todo list to speed up the learning process.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,15 @@
...
+ - [ ] make **browser-sync** task work
+ - [ ] figure out how to compress images and create a task for it
+ - [ ] figure out what is **autoprefixer** and update the sass task accordingly
+ - [ ] figure out what is **cache busting**, how it works and create a task for it
+ - [ ] figure out what is **babel** and adapt the ts compile task
+ - [ ] figure out what is **browserify** and adapt the ts compile task (add support for babel with babelify)
+ - [ ] figure out what **npm scripts** are and how they work
+ - [ ] create a build for development task
+ - [ ] create a build for production task (compress images, html, css, js)
+ - [ ] figure out what the **main** property from `package.json` does. Is it useful for front-end?
```
Comment (gleesik) [inline README.md:15]: Added a todo list to help the learning process.

File: gulpfile.js [hunk 2]
```diff
@@ -14,37 +7,32 @@ const browserSync = require('browser-sync');
...
 function compilePug() {
   return src(paths.pug + '*.pug')
-    .on('error', function (e) {
-      process.stderr.write(e.message + '\n');
-      this.emit('end');
-    })
-    .pipe(compilePug())
-    .pipe(dest(paths.htmlFolder));  
+    .pipe(pug())
+    .pipe(dest(paths.htmlFolder));
```
Comment (gleesik) [inline gulpfile.js:24]: Simplified the gulp task. Removed the recursive call.

File: gulpfile.js [hunk 3]
```diff
@@ -14,37 +7,32 @@ const browserSync = require('browser-sync');
...
 }
 
 function compileSass() {
   return src(paths.sass + '*.sass')
-    .on('error', sass.logError)
-    .pipe(compileSass())
+    .pipe(sass())
```
Comment (gleesik) [inline gulpfile.js:29]: Simplified the sass task. Removed useless code.

File: gulpfile.js [hunk 4]
```diff
@@ -54,9 +42,9 @@ function sync() {
 
 function watchFiles() {
   watch(paths.sass + '**/*.sass', series(compileSass))
-  .on('change', sync)  
+    .on('change', sync);
```
Comment (gleesik) [inline gulpfile.js:45]: Added ; 👩🏼‍🦯

File: package.json [hunk 5]
```diff
@@ -2,26 +2,15 @@
   "name": "onepager",
   "version": "1.0.0",
   "description": "Get familiar with Pug, Sass, Ts & Gulp",
-  "main": "./public/js/index.ts",
-  "dependencies": {
-    "gulp": "^4.0.2",
-    "typescript": "^3.9.7",
-    "sass": "^1.26.10"
-  },
+  "main": "",
```
Comment (gleesik) [inline package.json:5]: Removed the wrong pointing main start. Useful only for npm apps.

File: package.json [hunk 6]
```diff
@@ -2,26 +2,15 @@
...
+    "gulp": "^4.0.2",
     "gulp-pug": "^4.0.1",
-    "gulp-sass": "^4.1.0",
-    "mocha": "^8.0.1",
-    "pug": "^3.0.0"
+    "gulp-sass": "^4.1.0"
   },
   "scripts": {
-    "build": "build",
-    "default": "default",
```
Comment (gleesik) [inline package.json:24]: Removed useless dependencies. As less as possible is desired.

File: src/pug/_partials/_main.pug [hunk 7]
```diff
@@ -1,5 +1,5 @@
 //- Declaration (main.pug)
-mixin main
+mixin main()
```
Comment (gleesik) [inline src/pug/_partials/_main.pug:2]: Fixed mixin syntax.

File: src/pug/index.pug [hunk 8]
```diff
@@ -1,5 +1,9 @@
+include _partials/_head
+include _partials/_main
```
Comment (gleesik) [inline src/pug/index.pug:2]: Included the files that contain the used mixins.

File: README.md (hunk 1 above)
Comment (decameronn) [reply README.md:15]: This is really valuable advice. Thank you.

**Pull request merged**